packages/core/src/AttesterSchema.ts

```diff
@@ -0,0 +1,278 @@
+            const name = this.schema[i]['name']
+            const bits = BigInt(this.schema[i]['type'].slice(4))
+            const updateBy = this.schema[i]['updateBy']
+
+            if (updateBy == 'sum') {
+                if (
+                    !this.sumDataFields[this.curSumDataFieldCount].addOrPass(
+                        name,
+                        bits
+                    )
+                ) {
+                    this.curSumDataFieldCount++
+
+                    if (this.curSumDataFieldCount >= SUM_FIELD_COUNT) {
+                        throw Error('Too many sum fields')
+                    }
+                    const res = this.sumDataFields[
+                        this.curSumDataFieldCount
+                    ].addOrPass(name, bits)
+
+                    if (!res) {
+                        throw Error("Couldn't add " + name + ' sum data field')
+                    }
+                }
+            } else if (updateBy == 'replace') {
+                this.replDataFields[this.curReplDataFieldCount].setMaxFieldSize(
+                    254 - REPL_NONCE_BITS
+                )
+                if (bits > 254 - REPL_NONCE_BITS) {
+                    throw Error('Excessive replacement bits allocation')
+                }
+
+                if (
+                    !this.replDataFields[this.curReplDataFieldCount].addOrPass(
+                        name,
+                        bits
+                    )
+                ) {
+                    if (
+                        this.curReplDataFieldCount + SUM_FIELD_COUNT ==
+                        FIELD_COUNT
+                    ) {
+                        throw Error('Too many replacement and sum fields')
+                    }
+                    this.curReplDataFieldCount++
+
+                    const res = this.replDataFields[
+                        this.curReplDataFieldCount
+                    ].addOrPass(name, bits)
+
+                    if (!res) {
+                        throw Error(
+                            "Couldn't replace " +
+                                name +
+                                ' replacement data field'
+                        )
+                    }
+                }
+            }
+        }
+    }
+
+    getEncodedFieldIndex(entry: string): number {
+        const schema = this.findEntryByName(entry)
+        const updateBy = schema['updateBy']
+
+        if (updateBy == 'sum') {
+            for (let i = 0; i < SUM_FIELD_COUNT; i++) {
+                if (this.sumDataFields[i].field.includes(entry)) {
+                    return i
+                }
+            }
+        } else if (updateBy == 'replace') {
+            for (let i = 0; i < FIELD_COUNT - SUM_FIELD_COUNT; i++) {
+                if (this.replDataFields[i].field.includes(entry)) {
+                    return i
+                }
+            }
+        } else {
+            throw Error('Invalid update type')
```

Comment:
this could be tested

Reply:
Removed `Error()` for the same reasoning as for `update()`. Adding below for reference:

I realized that the error would never be called since 
```js
const schema = this.findEntryByName(entry)
```
would throw 
```js
throw Error('Entry does not exist in schema')
```
There are now two tests that check for this, one for `update()` and the other for `getEncodedFieldIndex()`